File: Day-1(basic)problem/Sum_Of_n/Sum_Of_n.py
def sum_of_n(n):
    sum=0
    for i in range(1,n+1):
        sum+=i
    return sum
if __name__ == "__main__":
    n=int(input("Enter a number: "))
    print(f"The sum of first {n} numbers is: {sum_of_n(n)}")
    #Time complexity: O(n) and Space complexity: O(1)

#Expecting using formula: O(1) time, O(1) space
def sum_of_n_formula(n):
    return n*(n+1)//2
if __name__ == "__main__":
    n=int(input("Enter a number: "))
    print(f"The sum of first {n} numbers is: {sum_of_n_formula(n)}")

File: Day-1(basic)problem/Sum_Of_n/Sum_of_n.c
#include <stdio.h>
void Sum_of_n(int n){
    int sum = 0;
    for(int i = 1;i<=n;i++){
        sum += i;
    }
    printf("Sum of first %d natural numbers is: %d\n", n, sum);
}
int main() {
    int n = 10;
    Sum_of_n(n);
    return 0;
}
// Time Complexity: O(n) and Space Complexity: O(1)

// Alternative Approach: Using formula n*(n+1)/2
int sum_of_n_formula(int n) {
    return n*(n+1)/2;
}
int main() {
    int n = 10;
    printf("Sum of first %d natural numbers is: %d\n", n, sum_of_n_formula(n));
    return 0;
}